dev/docker/doris/doris-dependency.sh

```diff
@@ -8,70 +8,108 @@ doris_dir="$(dirname "${BASH_SOURCE-$0}")"
+    doris_package_arch="${DORIS_ARM64}"
+  elif [[ "$arch" == "${AMD64}" ]]; then
+    doris_package_arch="${DORIS_X64}"
+  else
+    echo "ERROR : ${arch} is not a valid arch type for doris"
+    exit 1
+  fi
 
-if [[ ! -f "${doris_dir}/packages/${DORIS_FILE_NAME}" ]]; then
-  curl -s -o "${doris_dir}/packages/${DORIS_FILE_NAME}" ${DORIS_DOWNLOAD_URL}
-fi
+  # Download doris package
+  DORIS_PACKAGE_NAME="apache-doris-${DORIS_VERSION}-bin-${doris_package_arch}"
+  DORIS_FILE_NAME="${DORIS_PACKAGE_NAME}.tar.xz"
+  DORIS_DOWNLOAD_URL="https://apache-doris-releases.oss-accelerate.aliyuncs.com/${DORIS_FILE_NAME}"
+  SHA512SUMS_URL="${DORIS_DOWNLOAD_URL}.sha512"
 
-# download sha512sum file
-if [[ ! -f "${doris_dir}/packages/${DORIS_FILE_NAME}.sha512" ]]; then
-  curl -s -o "${doris_dir}/packages/${DORIS_FILE_NAME}.sha512" ${SHA512SUMS_URL}
-fi
+  download_dir="${doris_dir}/packages/"
 
+  # Prepare download packages
+  if [[ ! -d "${download_dir}" ]]; then
+    mkdir -p "${download_dir}"
+  fi
+
+  if [[ ! -f "${download_dir}/${DORIS_FILE_NAME}" ]]; then
+    echo "INFO : Downloading doris package ${download_dir}/${DORIS_FILE_NAME}"
+    curl -s -o "${download_dir}/${DORIS_FILE_NAME}" ${DORIS_DOWNLOAD_URL}
+    echo "INFO : Downloading doris package done"
+  fi
 
-cd "${doris_dir}/packages" || exit 1
+  # download sha512sum file
+  if [[ ! -f "${download_dir}/${DORIS_FILE_NAME}.sha512" ]]; then
+    curl -s -o "${download_dir}/${DORIS_FILE_NAME}.sha512" ${SHA512SUMS_URL}
+  fi
 
-# check sha512sum, if check file failed, exit 1
+  cd "${download_dir}" || exit 1
 
-if command -v shasum &>/dev/null; then
-  shasum -c "${DORIS_FILE_NAME}.sha512"
-elif command -v sha512sum &>/dev/null; then
-  sha512sum -c "${DORIS_FILE_NAME}.sha512"
-else
-  cat << EOF
-  WARN: cannot find shasum or sha512sum command, skip sha512sum check, please check the sha512sum by yourself.
-  if build Docker image failed, maybe the package is broken.
+  # check sha512sum, if check file failed, exit 1
+  echo "INFO : Checking sha512sum for ${DORIS_FILE_NAME}"
+  if command -v shasum &>/dev/null; then
+    shasum -c "${DORIS_FILE_NAME}.sha512"
+  elif command -v sha512sum &>/dev/null; then
+    sha512sum -c "${DORIS_FILE_NAME}.sha512"
+  else
+    cat << EOF
+    WARN: cannot find shasum or sha512sum command, skip sha512sum check, please check the sha512sum by yourself.
+    if build Docker image failed, maybe the package is broken.
 EOF
-fi
+  fi
+
+  if [ $? -ne 0 ]; then
+    echo "ERROR: check doris package sha512sum failed"
+    exit 1
+  fi
+
+  # copy doris package and rename to doris-${arch}.tar.xz
+  # it's a little trick, because the doris package name for 'amd64' is different as 'x64'
+  # rename to x64 can use ADD (instead of COPY) in Dockerfile, this will reduce the image size(about 1.6GB)
+  if [[ -f "${download_dir}/doris-${arch}.tar.xz" ]]; then
+    rm -f "${download_dir}/doris-${arch}.tar.xz"
+  fi
+
+  cp "${DORIS_FILE_NAME}" "doris-${arch}.tar.xz"
```

Comment:
is the current working dir here doris_dir`? I can't get why the file `"${DORIS_FILE_NAME}"` is under ${doris_dir} NOT `"${download_dir}"`

Reply:
in L59, we have `cd "${doris_dir}/packages" || exit 1`, so current dir is packages

Reply:
I see